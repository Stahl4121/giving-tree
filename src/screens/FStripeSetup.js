import React, { useEffect } from 'react';
import { withAuthProtection } from '../auth';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import Link from '@material-ui/core/Link';
import HelpOutlineIcon from '@material-ui/icons/HelpOutline';
import { makeStyles } from '@material-ui/core/styles';
import { Dialog, DialogContent, DialogActions, DialogTitle, DialogContentText, Slide, Grid } from '@material-ui/core';


const useStyles = makeStyles(theme => ({
  pageLayout: {
    width: 'auto',
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    [theme.breakpoints.up(600 + theme.spacing(2) * 2)]: {
      width: 600,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
    marginTop: theme.spacing(4),
    marginBottom: theme.spacing(2),
  },
}))

function FStripeSetup() {
  const classes = useStyles();

  // Set tab title
  useEffect(() => { document.title = 'Connect to Stripe' }, []);

  const [open, setOpen] = React.useState(false);
  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <div>
      <Grid container className={classes.pageLayout} justify='center' spacing={2}>
        <Grid item xs={10}>
          <Link
            textDecoration='none'
            color='inherit'
            href='https://connect.stripe.com/oauth/authorize?response_type=code&client_id=ca_GdKhmkaiFmOndJQ4UU6LTUm6pyfJcpQj&scope=read_write'>
            <Button
              fullWidth
              size="large"
              color='primary'
              variant='contained'>
              Connect to Stripe
            </Button>
          </Link>
        </Grid>
        <Grid item>
          <IconButton aria-label="Stripe Information" size="medium" onClick={handleClickOpen}>
            <HelpOutlineIcon />
          </IconButton>
          <Dialog
            open={open}
            TransitionComponent={Transition}
            keepMounted
            onClose={handleClose}
            aria-labelledby="alert-dialog-slide-title"
            aria-describedby="alert-dialog-slide-description"
          >
            <DialogTitle id="alert-dialog-slide-title">{"About Stripe"}</DialogTitle>
            <DialogContent>
              <DialogContentText id="alert-dialog-slide-description">
                <span>Stripe is a third party payment platform that handles customer payment information and allows donors to contribute to your
                  foundation's giving opportunities.&nbsp;</span>
                <span>After clicking on the link above, you will be taken to a signup page, where you will be prompted to provide your
                  foundation's basic information.&nbsp;</span>
                <span>Additionally, you will be given the opportunity to supply the information that will show up on the donors' banking and
                  credit card statements.&nbsp;</span>
                <span>Finally, you will be asked for your bank account number and routing number.&nbsp;</span>
                <span>Once all the pertinent information has been provided, you can strengthen your account's security with a second step of
                  authentication.&nbsp;</span>
                <span>If you still have questions about Stripe, please click "Learn More" below!</span>
              </DialogContentText>
            </DialogContent>
            <DialogActions>
              <Link
                textDecoration='none'
                color='inherit'
                href='https://stripe.com/about'
                target='_blank'
              >
                <Button onClick={handleClose} color="primary">
                  Learn More
                </Button>
              </Link>
              <Button onClick={handleClose} color="primary">
                Got it!
              </Button>
            </DialogActions>
          </Dialog>
        </Grid>
      </Grid>
    </div >
  );
}

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

export default withAuthProtection()(FStripeSetup);