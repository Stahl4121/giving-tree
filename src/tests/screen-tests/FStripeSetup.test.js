import React from 'react';
//import { shallow, mount } from 'enzyme';
import { createMount, createShallow } from '@material-ui/core/test-utils';

import FStripeSetup from '../../screens/FStripeSetup.js';

let mount;
let shallow;
beforeEach(() => {
  mount = createMount();
  shallow = createShallow();
});

describe('Links', () => {

  it('Renders correctly', () => {
    const wrap = shallow(<FStripeSetup />)
    expect(wrap).toMatchSnapshot();
  });

  it('Connect to Stripe link has correct address', () => {
    const wrap = mount(<FStripeSetup />)
    expect(wrap.find('ForwardRef(Link)').at(0).props().href).toEqual("https://connect.stripe.com/oauth/authorize?response_type=code&client_id=ca_GdKhmkaiFmOndJQ4UU6LTUm6pyfJcpQj&scope=read_write");
  });

  it('About link has correct address and opens new tab', () => {
    const wrap = mount(<FStripeSetup />)
    expect(wrap.find('ForwardRef(Link)').at(1).props().href).toEqual("https://stripe.com/about");
    expect(wrap.find('ForwardRef(Link)').at(1).props().target).toEqual('_blank');
  });

  
})

describe('Buttons', () => {
  it('They have proper text', () => {
    const wrap = mount(<FStripeSetup />)
    //console.log(wrap.find('ForwardRef(Button)').at(1).props())
    expect(wrap.find('ForwardRef(Button)').at(0).props().children).toEqual("Connect to Stripe");
    expect(wrap.find('ForwardRef(Button)').at(1).props().children).toEqual("Learn More");
    expect(wrap.find('ForwardRef(Button)').at(2).props().children).toEqual("Got it!");

    wrap.find('ForwardRef(IconButton)').at(0).simulate('click')
    expect(wrap.find('ForwardRef(Dialog)').at(0).props().open).toBeTruthy();

    wrap.find('ForwardRef(Button)').at(2).simulate('click')
    expect(wrap.find('ForwardRef(Dialog)').at(0).props().open).toBeFalsy();
  });

  it('The appropriate buttons open/close the dialog', () => {
    const wrap = mount(<FStripeSetup />)

    wrap.find('ForwardRef(IconButton)').at(0).simulate('click')
    expect(wrap.find('ForwardRef(Dialog)').at(0).props().open).toBeTruthy();

    wrap.find('ForwardRef(Button)').at(2).simulate('click')
    expect(wrap.find('ForwardRef(Dialog)').at(0).props().open).toBeFalsy();
  });
});