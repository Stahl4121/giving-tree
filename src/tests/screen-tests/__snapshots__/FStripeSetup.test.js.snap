// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links Renders correctly 1`] = `ShallowWrapper {}`;
